### PR TITLE
Fix: Add .idea to gitignore

### PR DESCRIPTION
**Pull request by innovationb1ue**
All IDE settings should be ignored

### CODE REVIEW

**Comment by GalvinGao** [comment]
Curious on any reference for why? I've pulled the .gitignore file from gitignore.io and it seems to have good reason on not simply ignoring all of them.

**Comment by innovationb1ue** [comment]
> Curious on any reference for why? I've pulled the .gitignore file from gitignore.io and it seems to have good reason on not simply ignoring all of them.

**Reason**: It doesn't ignore all the redundant project setting files in the .idea repository. 

I try open the project with Goland in both macOS and Windows. 
For example  `.idea/modules.xml`  `./idea/vcs.xml`  `./idea/backend-next.iml`  are popping up every time I open the project with Goland. It's neither useful for configuring the project nor for the IDE to function correctly. 
I have to manually discard the change with that file in order to not pollute the project with personal settings. 

I know that people must have a good reason for not simply ignoring all the files in .idea directory.  So I asked in the developer group chat. But seems all the answers were pointing to some kind of extension config and convenient IDE settings. That's not something need to be tracked with the project. 
Then I read the article whose link is included in the gitignore you pulled as a reference. You can find it here. 
[How to manage projects under Version Control Systems](https://intellij-support.jetbrains.com/hc/en-us/articles/206544839). 

It says that `.iml` files and `modules.xml` are **Items you may want to exclude from sharing**. And I am 100% sure that those files are not kept for any of the good reasons mentioned above. So I think either we should add those 3 files in gitignore since it will be automatically generated every single time I open the project with Goland, or we should just ignore all the files in `./idea` unless we really find something important has to be kept in that directory

**Comment by innovationb1ue** [comment]
> Curious on any reference for why? I've pulled the .gitignore file from gitignore.io and it seems to have good reason on not simply ignoring all of them.

It's like that we have already got this whole lot of lines for ignoring files in `.idea`. But still, some sneaky files get around the rules and polluting the project in a side way.  We should either fix it by adding lines or the whole folder to make it work like it should be, or that's no reason for those lines to be there as a flawed gitignore.

**Comment by FlandiaYingman** [comment]
References: 
 - https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
 - https://www.toptal.com/developers/gitignore/api/goland

tl; dr: no IDE settings in the repo; no IDE settings ignores in the project-level `.gitignore`

If we are in a same company and using the same IDE with the same rules, sharing the config files of IDE is somehow useful. By which we could conveniently share the formatter settings, run configurations, IDE inspections etc. But the choice of IDE is a preference, instead of some mandatory stuff. If we are not unifying the IDE we work with, then it's better to not include the settings - we always have replacements (`gofmt`, golint`, etc.).

So IMO, for an open-source project, we shouldn't include the IDE settings (whole `.idea` dir) into Git. (If we did so, the repo would eventually be saving everyone's settings)

However, just like I've said, the choice of IDE is a preference. So we neither include the IDE settings, nor add them to the project-level `.gitignore`. Instead, we add them to our user-level `.gitignore`. Through which, no matter what our preferred IDE is, the project-level `.gitignore` can keep unchanged.

**Comment by innovationb1ue** [comment]
> 

That's a really nice explanation. Lessons learned and thank you.
